receiver/k8seventsreceiver/k8s_event_to_logdata_test.go

```diff
@@ -51,6 +53,68 @@ func TestK8sEventToLogDataWithApiAndResourceVersion(t *testing.T) {
 	assert.Equal(t, "7387066320", attr.AsString())
 }
 
+func TestK8sEventToLogDataWithExtractionRules(t *testing.T) {
+	k8sEvent := getEvent()
+	k8sEvent.Labels["testlabel"] = "test-key-label"
+	k8sEvent.Labels["app.kubernetes.io/by-key"] = "test-key-label"
+	k8sEvent.Labels["app.kubernetes.io/by-regex"] = "test-regex-label"
+	k8sEvent.Annotations["topology.kubernetes.io/by-key"] = "test-key-annotation"
+	k8sEvent.Annotations["topology.kubernetes.io/by-regex"] = "test-regex-annotation"
+
+	labelsRules, err := ExtractFieldRules("labels",
+		FieldExtractConfig{
+			Key: "testlabel",
+		},
+		FieldExtractConfig{
+			TagName: "k8s.event.labels.label_by_key",
+			Key:     "app.kubernetes.io/by-key",
+		},
+		FieldExtractConfig{
+			TagName:  "k8s.event.labels.$1",
+			KeyRegex: "app.kubernetes.io/(.*)",
+		})
+	assert.NoError(t, err)
+
+	annotationsRules, err := ExtractFieldRules("annotations", FieldExtractConfig{
+		TagName: "k8s.event.annotations.annotation_by_key",
+		Key:     "topology.kubernetes.io/by-key",
+	}, FieldExtractConfig{
+		TagName:  "k8s.event.annotations.$1",
+		KeyRegex: "topology.kubernetes.io/(.*)",
+	})
+	assert.NoError(t, err)
+
+	ld := k8sEventToLogData(zap.NewNop(), k8sEvent, &kube.ExtractionRules{
+		Labels:      labelsRules,
+		Annotations: annotationsRules,
+	})
+	assert.NoError(t, err)
+
+	rl := ld.ResourceLogs().At(0)
+	lr := rl.ScopeLogs().At(0)
+	attrs := lr.LogRecords().At(0).Attributes()
+
+	attr, ok := attrs.Get("k8s.event.labels.testlabel")
```

Comment:
We will need to discuss and decide if `k8s.event.label/annotation` is the proper place for this. If so, I guess we would need to define those as Semantic Conventions.

/cc @TylerHelmuth @dmitryax

Reply:
We were planning to deprecate this receiver in favor of the k8sobjectsreceiver (https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24242). If the semconv is willing to define semantic conventions for k8s events we could consider reversing that decision, but my understanding if that we are no longer in the business of defining semconv for log attributes and are instead supposed to follow Otel's Event spec.

Reply:
I think focusing on https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24242 first would be ideal. It would not worth it adding stuff for k8sevents receiver if it's to get deprecated.